src/remote/routes.js

```diff
@@ -0,0 +1,242 @@
+      })
+    }
+  })
+
+  /**
+   * Start the daemon.
+   **/
+  server.route({
+    method: 'GET',
+    path: '/startDaemon',
+    handler: (request, reply) => {
+      const qr = parseQuery(request.query)
+      const id = qr.id
+      const flags = Object.values(qr.params ? qr.params.flags : {})
+      nodes[id].startDaemon(flags, (err) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, {
+          apiAddr: nodes[id].apiAddr.toString(),
+          gatewayAddr: nodes[id].gatewayAddr.toString()
+        }))
+      })
+    }
+  })
+
+  /**
+   * Stop the daemon.
+   */
+  server.route({
+    method: 'GET',
+    path: '/stopDaemon',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      nodes[id].stopDaemon((err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Kill the `ipfs daemon` process.
+   *
+   * First `SIGTERM` is sent, after 7.5 seconds `SIGKILL` is sent
+   * if the process hasn't exited yet.
+   */
+  server.route({
+    method: 'GET',
+    path: '/killProcess',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      nodes[id].killProcess((err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Get the pid of the `ipfs daemon` process.
+   *
+   * @returns {number}
+   */
+  server.route({
+    method: 'GET',
+    path: '/daemonPid',
+    handler: (request, reply) => {
+      const id = parseQuery(request.query).id
+      reply(makeResponse(id, nodes[id].daemonPid(nodes[id])))
+    }
+  })
+
+  /**
+   * Call `ipfs config`
+   *
+   * If no `key` is passed, the whole config is returned as an object.
+   */
+  server.route({
+    method: 'GET',
+    path: '/getConfig',
+    handler: (request, reply) => {
+      const qr = parseQuery(request.query)
+      const id = qr.id
+      const key = qr.params ? qr.params.key : null
+      nodes[id].getConfig(key, (err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Set a config value.
+   */
+  server.route({
+    method: 'GET',
+    path: '/setConfig',
+    handler: (request, reply) => {
+      const qr = parseQuery(request.query)
+      const id = qr.id
+      const key = qr.params ? qr.params.key : undefined
+      const val = qr.params ? qr.params.value : undefined
+      nodes[id].setConfig(key, val, (err, res) => {
+        if (err) {
+          return reply(boom.badRequest(err))
+        }
+
+        reply(makeResponse(id, res))
+      })
+    }
+  })
+
+  /**
+   * Replace the configuration with a given file
+   **/
+  server.route({
+    method: 'GET',
+    path: '/replaceConf',
```

Comment:
Shouldn't this handler be the same as the "partial" "set configuration" command above? You could use a query string param to indicate if it's to be replaced.

Example of a full replacement:

```
PUT /config?replace=true

{
  "all": "of",
  "the": "configs"
}
```

Example of a partial replacement:

```
PUT /config

{
  "one": "config value"
}
```

Reply:
The node id is missing from my example URLs, but the point remains.